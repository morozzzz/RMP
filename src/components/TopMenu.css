.top-menu {
    background: url('../assets/background_main.jpg');
    position: relative;
}

.top-menu-overlay {
    background-color: rgba(0, 0, 0, 0.644);
}

.top-menu-content-container {
    display: flex;
    justify-content: center;
} 

.top-menu-content {
    display: flex;
    flex-direction: column;
    align-items: flex-start;
    justify-content: space-between;
    width: 60%;
    margin: 1em 0;
}

.top-menu-item {
    margin-bottom: 1em;
}