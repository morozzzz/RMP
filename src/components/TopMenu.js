import React from 'react';
import SearchPannel from './SearchPannel';
import Logo from './Logo';

const TopMenu = () => (
    <div className="top-menu">
        <Logo />
        <SearchPannel />
    </div>
);
export default TopMenu;
