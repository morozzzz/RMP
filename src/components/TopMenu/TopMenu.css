.top-menu {
    background: url('../../assets/background_main.jpg');
}

.top-menu-overlay {
    background-color: rgba(0, 0, 0, 0.644);
    height: 100%;
}

.top-menu__content-container {
    display: flex;
    justify-content: center;
}

.top-menu__content {
    display: flex;
    flex-direction: column;
    align-items: flex-start;
    justify-content: space-between;
    width: 60%;
    margin: 1em 0;
}

.top-menu__item {
    margin-bottom: 1em;
}