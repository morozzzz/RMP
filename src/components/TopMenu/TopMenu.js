import React from 'react';
import SearchPanel from '../SearchPanel/SearchPanel';
import Logo from '../Logo/Logo';
import './TopMenu.css';

const TopMenu = ({ onSearchClick }) => (
    <div className="top-menu">
        <div className="top-menu-overlay">
            <div className="top-menu__content-container">
                <div className="top-menu__content">
                    <Logo name="netflixroulettee" redirectUrl="" className="top-menu__item" />
                    <SearchPanel onSearchClick={onSearchClick} className="top-menu__item" />
                </div>
            </div>
        </div>
    </div>
);

export default TopMenu;
