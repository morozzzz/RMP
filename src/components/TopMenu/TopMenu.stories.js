import React from 'react';
import { storiesOf } from '@storybook/react';
import { Provider } from 'react-redux';
import configureStore from '../../configureStore';
import TopMenu from './TopMenu';


const store = configureStore();

storiesOf('Top Menu', module)
    .addDecorator(story => <Provider store={store}>{story()}</Provider>)
    .add('Default', () => (
        <TopMenu />
    ));
