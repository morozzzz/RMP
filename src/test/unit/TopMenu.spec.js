import React from 'react';
import { shallow } from 'enzyme';
import TopMenu from '../../components/TopMenu/TopMenu';

jest.mock('../../components/Logo/Logo', () => () => <div id="Logo" />);
jest.mock('../../components/SearchPanel/SearchPanel', () => () => <div id="SearchPanel" />);

describe.skip('TopMenu', () => {
    it('should be rendered correctly', () => {
        const onSearchClickMock = () => {};
        const wrapper = shallow(<TopMenu onSearchClick={onSearchClickMock} />);

        expect(wrapper).toMatchSnapshot();
    });
});
