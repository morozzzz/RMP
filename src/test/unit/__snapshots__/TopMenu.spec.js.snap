// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TopMenu should be rendered correctly 1`] = `
<div
  className="top-menu"
>
  <div
    className="top-menu-overlay"
  >
    <div
      className="top-menu__content-container"
    >
      <div
        className="top-menu__content"
      >
        <Component
          className="top-menu__item"
          name="netflixroulettee"
          redirectUrl=""
        />
        <Component
          className="top-menu__item"
          onSearchClick={[Function]}
        />
      </div>
    </div>
  </div>
</div>
`;
